File: ProyectoVentas/src/entidades/dto/VentaDisplayDTO.java
package entidades.dto;

import java.time.LocalDateTime;
import java.util.List;
// El import de DetalleVentaDisplayDTO no es necesario si está en el mismo paquete.
// import entidades.dto.DetalleVentaDisplayDTO; // Se puede omitir

/**
 * DTO para mostrar la información de una venta de forma completa en la UI.
 * Incluye datos del administrador, detalles de productos y estado de la venta.
 */
public record VentaDisplayDTO(
    int idVenta,
    LocalDateTime fechaVenta,
    double montoTotal,
    String nombreAdministrador,
    List<DetalleVentaDisplayDTO> detalles,
    String estado
) {}

File: backend/src/types/VentaDisplayDTO.ts
import { EstadoVenta } from './Venta'; // Removed MetodoPago from import
import { DetalleVentaDisplayDTO } from './DetalleVentaDisplayDTO';

export interface VentaDisplayDTO {
  id_venta: number;
  fecha_venta: Date;
  total_venta: number;
  // metodo_pago: MetodoPago; // REMOVED
  id_administrador_venta?: number | null; 
  nombre_administrador_venta?: string | null; 
  estado: EstadoVenta; 
  detalles: DetalleVentaDisplayDTO[];
}
